source/fe/mapping_fe_field.cc

```diff
@@ -1043,10 +1043,11 @@ namespace internal
                                                       -1 : +1);
                   break;
                 case 2:
-                  cross_product (output_data.boundary_forms[i], data.aux[0][i]);
+                  output_data.boundary_forms[i] = cross_product(data.aux[0][i]);
```

Comment:
ah, here is where it would break down if you declared the one-argument function only for dim==2. so nevermind.